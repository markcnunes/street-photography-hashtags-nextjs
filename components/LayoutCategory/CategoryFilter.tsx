import React, { ChangeEvent, useState } from 'react';
import styled from '@emotion/styled';

export interface CategoryFilterProps {
  /**
   * Option of keywords available to be filtered.
   */
  options: string[];
  /**
   * Action to filter keywords.
   */
  filterKeywords: (arg: string[]) => void;
}

const CategoryFilter = ({ options, filterKeywords }: CategoryFilterProps) => {
  const [hidenCategories, setHidenCategories] = useState<string[]>([]);

  // Show or hide some specific category
  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    const name = event.target.name;
    let updateHidenCategories;

    if (event.target.checked) {
      // add
      updateHidenCategories = [...hidenCategories, name];
    } else {
      // remove
      updateHidenCategories = hidenCategories.filter((item) => item !== name);
    }
    filterKeywords(updateHidenCategories);
    setHidenCategories(updateHidenCategories);
  };

  // Show or hide all keywords
  const handleInputChangeAll = (event: ChangeEvent<HTMLInputElement>) => {
    const name = event.target.name;
    let updateHidenCategories: string[];

    if (name === 'all') {
      updateHidenCategories = [];
    } else {
      // if name = none
      updateHidenCategories = options;
    }

    filterKeywords(updateHidenCategories);
    setHidenCategories(updateHidenCategories);
  };

  return (
    <StyledCategoryFilter>
      <h4>Filter:</h4>
      <label
        className={
          hidenCategories.length === options.length ? 'active' : undefined
        }
      >
        Show All
        <input
          name='all'
          type='checkbox'
          checked={hidenCategories.length === options.length}
          onChange={handleInputChangeAll}
        />
      </label>
      <label className={hidenCategories.length === 0 ? 'active' : undefined}>
        Hide All
        <input
          name='none'
          type='checkbox'
          checked={hidenCategories.length === 0}
          onChange={handleInputChangeAll}
        />
      </label>
      {options.map((item) => (
        <label
          key={item}
          className={
            hidenCategories.some((category) => category === item)
              ? 'active'
              : undefined
          }
        >
          {item}
          <input
            name={item}
            type='checkbox'
            checked={hidenCategories.some((category) => category === item)}
            onChange={handleInputChange}
          />
        </label>
      ))}
    </StyledCategoryFilter>
  );
};

export default CategoryFilter;

const StyledCategoryFilter = styled('form')(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexWrap: 'wrap',
  width: '100%',
  padding: 10,
  fontSize: theme.fontSizes[1],
  color: theme.colors.white,
  border: `1px solid ${theme.colors.white}`,
  borderRadius: 5,
  marginBottom: 20,
  h4: {
    margin: 0,
    width: '100%',
    textAlign: 'center',
  },
  label: {
    display: 'flex',
    alignItems: 'center',
    padding: 15,
    lineHeight: 0,
    cursor: 'pointer',
    borderRadius: 3,
    margin: 5,
    textTransform: 'capitalize',
    '&.active': {
      opacity: 0.4,
    },
    '&:hover': {
      color: theme.colors.black,
      background: theme.colors.white,
    },
    input: {
      display: 'none',
    },
  },
  '@media (min-width: 780px)': {
    h4: {
      width: 'auto',
      lineHeight: 1,
      marginRight: 10,
    },
  },
}));
