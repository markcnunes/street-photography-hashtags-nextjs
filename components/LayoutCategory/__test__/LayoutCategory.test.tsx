import React from 'react';
import { render, screen } from 'utils/test-utils';
import LayoutCategory from '../LayoutCategory';
import mockData, { mockDataWichSubcategories } from '../__mock__/mockData';

const { category, icon, keywords, subcategories } = mockData;

describe('<LayoutCategory />', () => {
  it('matchs snapshot', () => {
    const { asFragment } = render(
      <LayoutCategory
        category={category}
        icon={icon}
        keywords={keywords}
        subcategories={subcategories}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders props', () => {
    render(
      <LayoutCategory
        category={category}
        icon={icon}
        keywords={keywords}
        subcategories={subcategories}
      />,
    );
    expect(screen.getByText(category)).toBeInTheDocument();
    expect(screen.getByTestId(icon)).toBeInTheDocument();
    expect(screen.getAllByTitle(/on Instagram/i)).toHaveLength(30);
  });
  it('show filter for category All', () => {
    const categoryAll = 'all';
    render(
      <LayoutCategory
        category={mockDataWichSubcategories.category}
        icon={mockDataWichSubcategories.icon}
        keywords={mockDataWichSubcategories.keywords}
        subcategories={mockDataWichSubcategories.subcategories}
      />,
    );
    expect(screen.getByText(/filter/i)).toBeInTheDocument();
  });
});
