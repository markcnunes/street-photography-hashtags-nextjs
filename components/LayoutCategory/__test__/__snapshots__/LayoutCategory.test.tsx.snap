// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutCategory /> matchs snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
}

.emotion-0 main {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  max-width: 90%;
  padding: 20px;
  margin: 0 auto;
}

.emotion-0 main>h4 {
  width: 100%;
  text-align: center;
  color: #fff;
}

.emotion-0 main>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
}

@media (min-width: 780px) {
  .emotion-0 main>div {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width: 1024px) {
  .emotion-0 main {
    max-width: 1200px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  border: 0;
  outline: 0;
  width: 100%;
  max-width: 90%;
  margin: 0 auto;
  color: #fff;
}

.emotion-2 .wrapper {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  width: 100%;
  padding: 20px;
  background: linear-gradient(10deg, rgba(255,255,255,0.1) 0%, rgba(230,230,230,0) 100%);
  border-radius: 5px;
}

.emotion-2 h3 {
  margin: 0 auto;
  text-transform: uppercase;
}

.emotion-2 a {
  position: absolute;
  left: 20px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2 a div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 .title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.emotion-2 .title svg {
  margin-right: 5px;
}

@media (max-width: 780px) {
  .emotion-2 .wrapper {
    padding-left: 40px;
    padding-right: 40px;
  }

  .emotion-2 a {
    left: 10px;
  }

  .emotion-2 a div {
    font-size: 0;
    padding: 10px;
  }

  .emotion-2 a div span {
    font-size: 16px;
    margin-right: 0;
  }
}

@media (min-width: 1024px) {
  .emotion-2 {
    max-width: 1200px;
  }
}

.emotion-5 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background: #fff;
  border-radius: 5px;
  width: 100%;
  padding: 20px;
  border-radius: 0 0 5px 5px;
  -webkit-flex: 1 0 0;
  -ms-flex: 1 0 0;
  flex: 1 0 0;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

@media (min-width: 1024px) {
  .emotion-5 {
    padding: 40px 30px;
  }
}

@media (min-width: 780px) {
  .emotion-5 {
    -webkit-order: 0;
    -ms-flex-order: 0;
    order: 0;
    border-radius: 5px 0 0 5px;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-right: 30px;
  min-height: 70px;
}

.emotion-7 a {
  color: #1c1c27;
  padding: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 5px;
  border: 1px solid transparent;
}

.emotion-7 a:hover {
  border-color: #536075;
}

.emotion-7:hover a {
  color: #536075;
}

.emotion-7:hover a:hover {
  color: #1c1c27;
}

@media (max-width: 420px) {
  .emotion-7 {
    font-size: 12px;
  }
}

.emotion-9 {
  position: absolute;
  top: 20px;
  right: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width: 1024px) {
  .emotion-9 {
    top: 40px;
  }
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28px;
  width: 28px;
  font-size: 10px;
  color: #1c1c27;
  border: 1px solid #1c1c27;
  border-radius: 5px;
  opacity: 0.5;
  margin-bottom: 10px;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: auto;
  -ms-flex-pack: auto;
  -webkit-justify-content: auto;
  justify-content: auto;
  border: 0;
  outline: 0;
  cursor: pointer;
  width: auto;
  opacity: 1;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  color: #fff;
  background: #1c1c27;
  font-size: 16px;
  padding: 8px 16px;
  border-radius: 4px;
  border: 1px solid currentColor;
  background: transparent;
  color: #1c1c27;
  padding: 5px;
}

.emotion-14:disabled {
  cursor: auto;
}

.emotion-14:hover {
  background: #474762;
}

.emotion-14:disabled {
  color: rgba(83,96,117,0.5);
}

.emotion-14:hover {
  background: rgba(28,28,39,0.05);
}

.emotion-14:hover:disabled {
  background: transparent;
  color: rgba(83,96,117,0.5);
}

.emotion-16 {
  color: #fff;
  background: rgba(28,28,39,0.5);
  border-radius: 5px 5px 0 0;
  width: 100%;
  padding: 20px;
}

.emotion-16>*+* {
  margin-top: 20px;
  padding-top: 20px;
  border-top: 1px solid #536075;
}

@media (min-width: 780px) {
  .emotion-16 {
    border-radius: 0 5px 5px 0;
    max-width: 280px;
  }
}

@media (min-width: 1024px) {
  .emotion-16 {
    padding: 40px;
    max-width: 360px;
  }
}

.emotion-18 h4 {
  margin-top: 0;
}

.emotion-18 form>* {
  width: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-18 label {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
  padding-left: 20px;
  border-radius: 5px;
  border: 1px solid currentColor;
}

.emotion-18 label::before,
.emotion-18 label::after {
  content: "";
  border-right: 4px solid transparent;
  border-left: 4px solid transparent;
  border-top: 4px solid #536075;
  position: absolute;
  top: 50%;
  right: 15px;
  pointer-events: none;
}

.emotion-18 label::before {
  -webkit-transform: translateY(-180%) rotate(180deg);
  -moz-transform: translateY(-180%) rotate(180deg);
  -ms-transform: translateY(-180%) rotate(180deg);
  transform: translateY(-180%) rotate(180deg);
}

.emotion-18 label::after {
  -webkit-transform: translateY(-20%);
  -moz-transform: translateY(-20%);
  -ms-transform: translateY(-20%);
  transform: translateY(-20%);
}

.emotion-18 label select {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin-left: auto;
  padding: 10px 45px 10px 35px;
  background: #fff;
  border: none;
  outline: none;
}

.emotion-18 button svg {
  width: 25px;
  height: 25px;
}

.emotion-20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: auto;
  -ms-flex-pack: auto;
  -webkit-justify-content: auto;
  justify-content: auto;
  border: 0;
  outline: 0;
  cursor: pointer;
  width: auto;
  opacity: 1;
  border-radius: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.3s ease;
  transition: background 0.3s ease;
  color: #fff;
  background: #615cf5;
  font-size: 16px;
  padding: 8px 16px;
  border-radius: 4px;
}

.emotion-20 svg,
.emotion-20 span {
  margin-right: 5px;
}

.emotion-20:disabled {
  cursor: auto;
}

.emotion-20:hover {
  background: #706bff;
}

.emotion-20:disabled {
  background: rgba(83,96,117,0.5);
  color: #fff;
}

<div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="wrapper"
      >
        <div
          class="title"
        >
          <svg
            data-testid="GiSpy"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M218 19c-1 0-2.76.52-5.502 3.107-2.742 2.589-6.006 7.021-9.191 12.76-6.37 11.478-12.527 28.033-17.666 45.653-4.33 14.844-7.91 30.457-10.616 44.601 54.351 24.019 107.599 24.019 161.95 0-2.706-14.144-6.286-29.757-10.616-44.601-5.139-17.62-11.295-34.175-17.666-45.653-3.185-5.739-6.45-10.171-9.191-12.76C296.76 19.52 295 19 294 19c-6.5 0-9.092 1.375-10.822 2.85-1.73 1.474-3.02 3.81-4.358 7.34-1.338 3.53-2.397 8.024-5.55 12.783C270.116 46.73 263.367 51 256 51c-7.433 0-14.24-4.195-17.455-8.988-3.214-4.794-4.26-9.335-5.576-12.881-1.316-3.546-2.575-5.867-4.254-7.315C227.035 20.37 224.5 19 218 19zm-46.111 124.334c-1.41 9.278-2.296 17.16-2.57 22.602 6.61 5.087 17.736 10.007 31.742 13.302C217.18 183.031 236.6 185 256 185s38.82-1.969 54.94-5.762c14.005-3.295 25.13-8.215 31.742-13.302-.275-5.443-1.161-13.324-2.57-22.602-55.757 23.332-112.467 23.332-168.223 0zM151.945 155.1c-19.206 3.36-36.706 7.385-51.918 11.63-19.879 5.548-35.905 11.489-46.545 16.57-5.32 2.542-9.312 4.915-11.494 6.57-.37.28-.247.306-.445.546.333.677.82 1.456 1.73 2.479 1.973 2.216 5.564 4.992 10.627 7.744 10.127 5.504 25.944 10.958 45.725 15.506C139.187 225.24 194.703 231 256 231s116.813-5.76 156.375-14.855c19.78-4.548 35.598-10.002 45.725-15.506 5.063-2.752 8.653-5.528 10.627-7.744.91-1.023 1.397-1.802 1.73-2.479-.198-.24-.075-.266-.445-.547-2.182-1.654-6.174-4.027-11.494-6.568-10.64-5.082-26.666-11.023-46.545-16.57-15.212-4.246-32.712-8.272-51.918-11.631.608 5.787.945 10.866.945 14.9v3.729l-2.637 2.634c-10.121 10.122-25.422 16.191-43.302 20.399C297.18 200.969 276.6 203 256 203s-41.18-2.031-59.06-6.238c-17.881-4.208-33.182-10.277-43.303-20.399L151 173.73V170c0-4.034.337-9.113.945-14.9zm1.094 88.205C154.558 308.17 200.64 359 256 359c55.36 0 101.442-50.83 102.96-115.695a748.452 748.452 0 0 1-19.284 2.013c-1.33 5.252-6.884 25.248-15.676 30.682-13.61 8.412-34.006 7.756-48 0-7.986-4.426-14.865-19.196-18.064-27.012-.648.002-1.287.012-1.936.012-.65 0-1.288-.01-1.936-.012-3.2 7.816-10.078 22.586-18.064 27.012-13.994 7.756-34.39 8.412-48 0-8.792-5.434-14.346-25.43-15.676-30.682a748.452 748.452 0 0 1-19.285-2.013zM137.4 267.209c-47.432 13.23-77.243 32.253-113.546 61.082 42.575 4.442 67.486 21.318 101.265 48.719l16.928 13.732-21.686 2.211c-13.663 1.393-28.446 8.622-39.3 17.3-5.925 4.738-10.178 10.06-12.957 14.356 44.68 5.864 73.463 10.086 98.011 20.147 18.603 7.624 34.81 18.89 53.737 35.781l5.304-23.576c-1.838-9.734-4.134-19.884-6.879-30.3-5.12-7.23-9.698-14.866-13.136-22.007C201.612 397.326 199 391 199 384c0-3.283.936-6.396 2.428-9.133a480.414 480.414 0 0 0-6.942-16.863c-29.083-19.498-50.217-52.359-57.086-90.795zm237.2 0c-6.87 38.436-28.003 71.297-57.086 90.795a480.521 480.521 0 0 0-6.942 16.861c1.493 2.737 2.428 5.851 2.428 9.135 0 7-2.612 13.326-6.14 20.654-3.44 7.142-8.019 14.78-13.14 22.01-2.778 10.547-5.099 20.82-6.949 30.666l5.14 23.42c19.03-17.01 35.293-28.338 53.974-35.994 24.548-10.06 53.33-14.283 98.011-20.147-2.78-4.297-7.032-9.618-12.957-14.355-10.854-8.679-25.637-15.908-39.3-17.3l-21.686-2.212 16.928-13.732c33.779-27.4 58.69-44.277 101.265-48.719-36.303-28.829-66.114-47.851-113.546-61.082zM256 377c-8 0-19.592.098-28.234 1.826-4.321.864-7.8 2.222-9.393 3.324-1.592 1.103-1.373.85-1.373 1.85s1.388 6.674 4.36 12.846c2.971 6.172 7.247 13.32 11.964 19.924 4.717 6.604 9.925 12.699 14.465 16.806 4.075 3.687 7.842 5.121 8.211 5.377.37-.256 4.136-1.69 8.21-5.377 4.54-4.107 9.749-10.202 14.466-16.806 4.717-6.605 8.993-13.752 11.965-19.924C293.612 390.674 295 385 295 384s.22-.747-1.373-1.85c-1.593-1.102-5.072-2.46-9.393-3.324C275.592 377.098 264 377 256 377zm0 61.953c-.042.03-.051.047 0 .047s.042-.018 0-.047zm-11.648 14.701L235.047 495h41.56l-9.058-41.285C264.162 455.71 260.449 457 256 457c-4.492 0-8.235-1.316-11.648-3.346z"
            />
          </svg>
          <h3>
            candid
          </h3>
        </div>
      </div>
    </div>
    <main>
      <div>
        <div
          class="emotion-4 emotion-5 emotion-6"
        >
          <div
            class="emotion-7 emotion-8"
          >
            <a
              href="https://www.instagram.com/explore/tags/test1/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test1 on Instagram"
            >
              #test1
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test2/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test2 on Instagram"
            >
              #test2
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test3/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test3 on Instagram"
            >
              #test3
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test4/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test4 on Instagram"
            >
              #test4
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test5/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test5 on Instagram"
            >
              #test5
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test6/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test6 on Instagram"
            >
              #test6
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test7/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test7 on Instagram"
            >
              #test7
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test8/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test8 on Instagram"
            >
              #test8
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test9/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test9 on Instagram"
            >
              #test9
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test10/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test10 on Instagram"
            >
              #test10
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test11/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test11 on Instagram"
            >
              #test11
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test12/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test12 on Instagram"
            >
              #test12
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test13/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test13 on Instagram"
            >
              #test13
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test14/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test14 on Instagram"
            >
              #test14
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test15/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test15 on Instagram"
            >
              #test15
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test16/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test16 on Instagram"
            >
              #test16
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test17/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test17 on Instagram"
            >
              #test17
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test18/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test18 on Instagram"
            >
              #test18
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test19/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test19 on Instagram"
            >
              #test19
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test20/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test20 on Instagram"
            >
              #test20
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test21/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test21 on Instagram"
            >
              #test21
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test22/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test22 on Instagram"
            >
              #test22
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test23/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test23 on Instagram"
            >
              #test23
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test24/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test24 on Instagram"
            >
              #test24
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test25/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test25 on Instagram"
            >
              #test25
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test26/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test26 on Instagram"
            >
              #test26
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test27/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test27 on Instagram"
            >
              #test27
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test28/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test28 on Instagram"
            >
              #test28
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test29/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test29 on Instagram"
            >
              #test29
            </a>
            <a
              href="https://www.instagram.com/explore/tags/test30/"
              rel="noopener noreferrer"
              target="_blank"
              title="View test30 on Instagram"
            >
              #test30
            </a>
          </div>
          <div
            class="emotion-9 emotion-10"
          >
            <div
              class="emotion-11 emotion-12"
            >
              30
            </div>
            <button
              class="emotion-13 emotion-14 emotion-15"
              color="black"
              title="Copy all to clipboard"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2"
                />
                <rect
                  height="4"
                  rx="1"
                  ry="1"
                  width="8"
                  x="8"
                  y="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="emotion-16 emotion-17"
        >
          <div
            class="emotion-18 emotion-19"
          >
            <h4>
              Random selection of keywords
            </h4>
            <form>
              <label>
                Quantity
                <select>
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9
                  </option>
                  <option
                    value="10"
                  >
                    10
                  </option>
                  <option
                    value="11"
                  >
                    11
                  </option>
                  <option
                    value="12"
                  >
                    12
                  </option>
                  <option
                    value="13"
                  >
                    13
                  </option>
                  <option
                    value="14"
                  >
                    14
                  </option>
                  <option
                    value="15"
                  >
                    15
                  </option>
                  <option
                    value="16"
                  >
                    16
                  </option>
                  <option
                    value="17"
                  >
                    17
                  </option>
                  <option
                    value="18"
                  >
                    18
                  </option>
                  <option
                    value="19"
                  >
                    19
                  </option>
                  <option
                    value="20"
                  >
                    20
                  </option>
                  <option
                    value="21"
                  >
                    21
                  </option>
                  <option
                    value="22"
                  >
                    22
                  </option>
                  <option
                    value="23"
                  >
                    23
                  </option>
                  <option
                    value="24"
                  >
                    24
                  </option>
                  <option
                    value="25"
                  >
                    25
                  </option>
                  <option
                    value="26"
                  >
                    26
                  </option>
                  <option
                    value="27"
                  >
                    27
                  </option>
                  <option
                    value="28"
                  >
                    28
                  </option>
                  <option
                    value="29"
                  >
                    29
                  </option>
                  <option
                    value="30"
                  >
                    30
                  </option>
                </select>
              </label>
              <button
                class="emotion-20 emotion-15"
                color="primary"
                title="Generate a random limited selection of keywords"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M255.76 44.764c-6.176 0-12.353 1.384-17.137 4.152L85.87 137.276c-9.57 5.536-9.57 14.29 0 19.826l152.753 88.36c9.57 5.536 24.703 5.536 34.272 0l152.753-88.36c9.57-5.535 9.57-14.29 0-19.825l-152.753-88.36c-4.785-2.77-10.96-4.153-17.135-4.153zm-117.313 82.61a31.953 18.96 0 0 1 .002 0 31.953 18.96 0 0 1 21.195 5.536 31.953 18.96 0 0 1-45.19 26.813 31.953 18.96 0 0 1 23.992-32.348zm118.24.245a31.953 18.96 0 0 1 22.125 32.362 31.953 18.96 0 1 1-45.187-26.812 31.953 18.96 0 0 1 23.06-5.55zm119.663.015a31.953 18.96 0 0 1 .002 0 31.953 18.96 0 0 1 21.195 5.535 31.953 18.96 0 0 1-45.19 26.812 31.953 18.96 0 0 1 23.993-32.347zM75.67 173.84c-5.753-.155-9.664 4.336-9.664 12.28v157.696c0 11.052 7.57 24.163 17.14 29.69l146.93 84.848c9.57 5.526 17.14 1.156 17.14-9.895V290.76c0-11.052-7.57-24.16-17.14-29.688l-146.93-84.847c-2.69-1.555-5.225-2.327-7.476-2.387zm360.773.002c-2.25.06-4.783.83-7.474 2.385l-146.935 84.847c-9.57 5.527-17.14 18.638-17.14 29.69v157.7c0 11.05 7.57 15.418 17.14 9.89L428.97 373.51c9.57-5.527 17.137-18.636 17.137-29.688v-157.7c0-7.942-3.91-12.432-9.664-12.278zm-235.146 86.592a31.236 18.008 58.094 0 1 33.818 41.183 31.236 18.008 58.094 1 1-45-25.98 31.236 18.008 58.094 0 1 11.182-15.203zM366.82 289.1a18.008 31.236 31.906 0 1 .002 0 18.008 31.236 31.906 0 1 11.18 15.203 18.008 31.236 31.906 0 1-45 25.98A18.008 31.236 31.906 0 1 366.82 289.1zM89.297 318.48a31.236 18.008 58.094 0 1 33.818 41.184 31.236 18.008 58.094 1 1-45-25.98 31.236 18.008 58.094 0 1 11.182-15.204z"
                  />
                </svg>
                I'm Feeling Lucky
              </button>
            </form>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
